lib/influxer/metrics/metrics.rb

```diff
@@ -162,10 +170,50 @@ def tags
       @attributes.select { |k, _| tag_names.include?(k.to_s) }
     end
 
-    attributes :timestamp
+    def time(parse = false)
+      return unless @time
+
+      return parse_timestamp(@time) if parse
+      @time
+    end
+
+    def time=(val)
+      if val.is_a?(Date) || val.is_a?(Time)
+        @time = parse_time(val)
+      else
+        @time
+      end
+    end
 
     private
 
+    def parse_timestamp(val)
+      case client.time_precision
+      when 'ms'
```

Comment:
I agree that we don't have to support all possible precisions. Hence we should choose the only one for transforming – 'ms'.

And then it can be smth like this:

```ruby
def parse_timestamp(val)
  case val
  when Number
    val
  when String
    Time.parse(val).to_f * 1000
  when Date
    val.to_time.to_f  * 1000
  when Time
    val.to_f * 1000
  end
end
```